src/de/deserializer.zig

```diff
@@ -580,6 +524,10 @@ fn Union(comptime D: type) type {
     return struct {
         d: *D,
 
+        // TODO: allow returning allocated variants from variantSeed
+        /// A place to temporarily store heap-allocated variants.
+        variant_buf: [512]u8 = undefined,
```

Comment:
That would be a memory leak. The reason I copy it to the buffer is in order to be able to free it afterwards (done by `defer self.d.freeToken` here. If I were to return it directly, there would be no way to free the token at all, and since the buffer here is a part of the struct, it's on the stack.

Reply:
Oh, I see now. I thought you were freeing it elsewhere but no it's freed in `variantSeed` itself.

Makes sense!